legal-api/src/legal_api/services/permissions.py

```diff
@@ -0,0 +1,201 @@
+# Copyright © 2025 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# pylint: disable=too-many-lines
+"""This manages all of the authentication and authorization service."""
+from enum import Enum
+from http import HTTPStatus
+
+from flask import jsonify
+from flask import current_app, g
+
+from urllib3.util.retry import Retry
+
+from legal_api.models.authorized_role_permission import AuthorizedRolePermission
+from legal_api.services.cache import cache
+
+class ListFilingsPermissionsAllowed(str, Enum):
+    """Define an enum for permissions checks."""
+
+    STAFF_FILINGS = 'STAFF_FILINGS'
+    TRANSITION_FILING = 'TRANSITION_FILING'
+    VOLUNTARY_DISSOLUTION_FILING = 'VOLUNTARY_DISSOLUTION_FILING'
+    ADMIN_DISSOLUTION_FILING = 'ADMIN_DISSOLUTION_FILING'
```

Comment:
NB - we don't have a permission for Involuntary Dissolution (and that's OK).

Reply:
I'm resolving this conversation. Please reopen it if you find anything wrong or that needs to be addressed.